File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/ui/processing/ProcessingActivity.java
package com.thibaudperso.sonycamera.timelapse.ui.processing;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;

import com.thibaudperso.sonycamera.R;
import com.thibaudperso.sonycamera.timelapse.ui.SingleFragmentActivity;


public class ProcessingActivity extends SingleFragmentActivity {

    public static final String ACTION_STOP = "stop";

    private ProcessingFragment mFragment;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mEnableStateMachineConnection = false;
    }

    @Override
    protected Fragment createFragment() {
        mFragment = new ProcessingFragment();
        return mFragment;
    }

    @Override
    protected String getGuideTitle() {
        return getString(R.string.title_processing);
    }

    @Override
    public void onBackPressed() {

        if(mFragment == null || mFragment.isFinished()) {
            super.onBackPressed();
            return;
        }

        mFragment.askToStopProcessing(new ProcessingFragment.TimelapseStopListener() {
            @Override
            public void onStop() {
                ProcessingActivity.super.onBackPressed();
            }
        });
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);

        if (ACTION_STOP.equals(intent.getAction())) {
            mFragment.askToStopProcessing();
        }
    }
}

File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/ui/processing/ProcessingActivity.kt
package com.thibaudperso.sonycamera.timelapse.ui.processing

import android.content.Intent
import android.os.Bundle
import com.thibaudperso.sonycamera.R
import com.thibaudperso.sonycamera.timelapse.ui.SingleFragmentActivity
import mu.KLogging

class ProcessingActivity : SingleFragmentActivity<ProcessingFragment>() {

    companion object : KLogging() {
        const val ACTION_STOP = "stop"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableStateMachineConnection = false
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)

        if (ACTION_STOP == intent.action) {
            fragment.askToStopProcessing()
        }
    }

    override fun onBackPressed() {
        if (fragment.isFinished) {
            super.onBackPressed()
            return
        }

        fragment.askToStopProcessing { super@ProcessingActivity.onBackPressed() }
    }

    override fun createFragment() = ProcessingFragment()

    override val guideTitle: String
        get() = getString(R.string.title_processing)

}
